src/plugins/visualizations/public/persisted_state/persisted_state.ts

```diff
@@ -236,7 +236,7 @@ export class PersistedState extends EventEmitter {
     }
 
     // update the merged state value
-    const targetObj = this._mergedState || cloneDeep(this._defaultState);
+    const targetObj = cloneDeep(this._mergedState) || cloneDeep(this._defaultState);
```

Comment:
Did this used to fail because we were updating the same object after the first time, so reference inequality checks would fail and not update the view?

Reply:
@virajsanghvi The operation to update the newValue after the first update was bugging out, although the values being passed down were correct the update operation wasn't able set it to correctly. Unfortunately despite analyzing in detail I am unable to discover the details of how the operation was getting bugged out. But definitely will keep you posted in case I am able to get the details.

Reply:
What caused you to clone the object?